pkg/api/chunk.go

```diff
@@ -92,16 +100,18 @@ func (s *server) chunkUploadHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	// Indicate that the chunk is stored
-	err = tag.Inc(tags.StateStored)
-	if err != nil {
-		s.Logger.Debugf("chunk upload: increment tag", err)
-		s.Logger.Error("chunk upload: increment tag")
-		jsonhttp.BadRequest(w, "increment tag")
-		return
+	if tag != nil {
+		// indicate that the chunk is stored
+		err = tag.Inc(tags.StateStored)
+		if err != nil {
+			s.Logger.Debugf("chunk upload: increment tag", err)
+			s.Logger.Error("chunk upload: increment tag")
+			jsonhttp.BadRequest(w, "increment tag")
```

Comment:
As above, I would use server error as it is not related to the user input. It looks like that it was a mistake from earlier implementation carried here.